site/components/YoutubeEmbed/index.tsx

```diff
@@ -0,0 +1,102 @@
+import { FC, HTMLAttributes, useState } from "react";
+import { toNumber } from "lodash";
+import * as styles from "./styles";
+import Image, { ImageProps } from "next/image";
+
+/**
+ * For more information about the available youtube embed parameters see:
+ * https://developers.google.com/youtube/player_parameters#release_notes_08_23_2018
+ */
+type YoutubeEmbedProps = {
+  videoId: string;
+  autoplay?: boolean;
+  showSubtitles?: boolean;
+  hideControls?: boolean;
+  hideFullscreen?: boolean;
+  posterImg?: {
+    src: ImageProps["src"];
+    alt: string;
+  };
+} & HTMLAttributes<HTMLDivElement>;
+
+/**
+ * A wrapper around a youtube IFrame to provide a poster overlay
+ * @note auto play does not function on mobile devices
+ * see: https://stackoverflow.com/questions/15090782/youtube-autoplay-not-working-on-mobile-devices-with-embedded-html5-player
+ */
+export const YoutubeEmbed: FC<YoutubeEmbedProps> = ({
+  videoId,
+  posterImg,
+  autoplay,
+  showSubtitles,
+  hideControls,
+  hideFullscreen,
+  ...props
+}) => {
```

Comment:
https://github.com/KlimaDAO/klimadao/wiki#do-not-destructure-props/

Unless you need to set default values, which it looks like you arent doint in this case

Reply:
Sure thing